src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/utils/RangeUtilsTest.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.homeoffice.digital.sas.balancecalculator.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.homeoffice.digital.sas.balancecalculator.utils.RangeUtils.splitOverDays;
+
+import com.google.common.collect.Range;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.util.SortedMap;
+import org.junit.jupiter.api.Test;
+
+class RangeUtilsTest {
+
+  @Test
+  void splitOverDays_timeEntryWithinCalendarDay_returnOneDateTimeRange() {
```

Comment:
I thinks there's a good level of scenarios i.e. 1 day vs 2 day but I think more test cases would still be beneficial. Even if it was just one additional test
```
@ParameterizedTest
@MethodSource("testData")
void splitCorrectly(start, end, expectedRanges, endpoints[]) {
  ranges = splitOverDays(start, end)
  ranges.hasLength(expectedRanges)
  for (x = 0; x < expectedRanges * 2; x = x + 2) {
    assertThat(range[x].lower).isEqualTo(endpoints[x]);
    assertThat(range[x].upper).isEqualTo(endpoints[x + 1]);
  }
}
```
Then pump it with test cases including mismatched timezones, crossovers between GMT BST etc. If I understand the code it should sanitise all of these into Europe/London

Reply:
Updated the RangeUtils tests now to use a parameterised test. We also added extra scenarios for timezones and crossovers between BST and GMT

Reply:
Looks good. Thank you